src/inmanta/deploy/persistence.py

```diff
@@ -289,3 +310,65 @@ def get_resource_action_logger(self, environment: UUID) -> logging.Logger:
         :return: The logger for the given environment.
         """
         return self._resource_action_logger.getChild(str(environment))
+
+    async def dryrun_update(self, env: UUID, dryrun_result: DryrunResult) -> None:
+        # async with self.dryrun_lock: Should this class have this lock or one of the child classes?
+        payload = {
+            "changes": dryrun_result.changes,
+            "id_fields": Id.parse_id(dryrun_result.rvid).to_dict(),
+            "id": dryrun_result.rvid,
+        }
+        await data.DryRun.update_resource(dryrun_result.dryrun_id, dryrun_result.rvid, payload)
+
+        async with data.Resource.get_connection() as inner_connection:
+            async with inner_connection.transaction():
+                # validate resources
+                resource_id_parsed = Id.parse_id(dryrun_result.rvid)
+                version = resource_id_parsed.version
+                resource = await data.Resource.get_one(
+                    environment=env,
+                    resource_id=resource_id_parsed.resource_str(),
+                    model=version,
+                    # acquire lock on Resource before read and before lock on ResourceAction to prevent conflicts with
+                    # cascading deletes
+                    lock=data.RowLockMode.FOR_NO_KEY_UPDATE,
+                    connection=inner_connection,
+                )
+                assert (
+                    resource is not None
+                ), f"Resource {resource_id_parsed} does not exists in the database, this should not happen"
```

Comment:
We could probably do without checking the resource, but we can leave it in for now